bentoml/cli/aws_ec2.py

```diff
@@ -0,0 +1,376 @@
+                describe_result.status
+            )
+            raise CLIException(f"{error_code}:{error_message}")
+
+        get_result = yatai_client.deployment.get(namespace, name)
+        if get_result.status.status_code != yatai_proto.status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                describe_result.status
+            )
+            raise CLIException(f"{error_code}:{error_message}")
+
+        _print_deployment_info(get_result.deployment, output)
+
+    @aws_ec2.command(help="Update existing AWS EC2 deployments")
+    @click.argument("name", type=click.STRING)
+    @click.option(
+        "-b",
+        "--bento",
+        "--bento-service-bundle",
+        type=click.STRING,
+        callback=parse_bento_tag_callback,
+        help="Target BentoService to be deployed, referenced by its name and version "
+        'in format of name:version. For example: "iris_classifier:v1.2.0"',
+    )
+    @click.option(
+        "-n",
+        "--namespace",
+        type=click.STRING,
+        help='Deployment namespace managed by BentoML, default value is "dev" which '
+        'can be changed in BentoML configuration yatai_service/default_namespace',
+    )
+    @click.option(
+        "--min-capacity",
+        type=click.INT,
+        default=DEFAULT_MIN_CAPACITY,
+        help="The minimum limit helps ensure that you always have a "
+        "certain number of instances running at all times.Default is 1",
+    )
+    @click.option(
+        "--desired-capacity",
+        type=click.INT,
+        default=DEFAULT_DESIRED_CAPACITY,
+        help="Desired number of instances capacity to run BentoService on."
+        "Should be between minimum and maximum capacities.Default is 1",
+    )
+    @click.option(
+        "--max-capacity",
+        type=click.INT,
+        default=DEFAULT_MAX_CAPACITY,
+        help="The maximum limit lets Amazon EC2 Auto Scaling scale out "
+        "the number of instances as needed to handle an increase in demand. "
+        "Default is 1",
+    )
+    @click.option(
+        "--instance-type",
+        type=click.STRING,
+        default=DEFAULT_INSTANCE_TYPE,
+        help="Instance type of EC2 container.Default is t2 micro",
+    )
+    @click.option(
+        "--ami-id",
+        type=click.STRING,
+        default=DEFAULT_AMI_ID,
+        help="AMI id.Default is Amazon Linux 2",
+    )
+    @click.option(
+        "-o", "--output", type=click.Choice(["json", "yaml", "table"]), default="json"
+    )
+    @click.option(
+        "--wait/--no-wait",
+        default=True,
+        help="Wait for apply action to complete or encounter an error."
+        "If set to no-wait, CLI will return immediately. The default value is wait",
+    )
+    def update(
+        name,
+        bento,
+        namespace,
+        min_capacity,
+        desired_capacity,
+        max_capacity,
+        instance_type,
```

Comment:
just curious, how hard is it to add support for AWS spot instance? what the API would look like if we were to support it?

Reply:
Btw, no need to support it in this PR, just a quick question in case you have looked into it

Reply:
There are few distribution strategies for on-demand and spot with autoscaling group based on different parameters like price,usage,weights.Api could be designed to accept base on-demand instances and portion of on-demand and spot above base if mroe scaling-out required.